### PR TITLE
[2.x] feat: remove `behat/mink` requirement and add new selector API

### LINKED ISSUE #12: $selector ambiguity
The method parameter `$selector` is used in several methods and means different things depending on the context:

- *css selector*: `->assertSeeIn('h1', 'some text')` or `->dump('h1')`
- *link selector (link id/title/text/image alt)*: `->follow('some link')`
- *button selector (button id/value/alt)*: `->click('button')`
- *field selector (input id/name/label)*: `->fillField('name', 'Kevin')`
- *json search expression*: `->assertJsonMatches('foo.bar.baz', 1)` or `->dump('foo.bar.baz')` (when content-type is json)

I think adding good docblocks would help but there are some things that can be improved:

Right now, methods that select elements with link/button/field selectors cannot use css selectors to find the element. I think, if unable to find the element via the link/button/field selector, assume a css selector and try.

This opens up the possibility of clicking buttons via `tr:contains(First Post) a.btn` (think admin list of entities each with an edit button).

### LINKED ISSUE #37: Xpath support
```php
$this->wrapMinkExpectation(
     fn() => $this->webAssert()->elementExists('xpath', $xpathString)
);
```

### PR DESCRIPTION
**Pull request by kbond**
Fixes https://github.com/zenstruck/browser/pull/15, Fixes https://github.com/zenstruck/browser/issues/12. Fixes https://github.com/zenstruck/browser/issues/37. Fixes #77.

Replacing mink with my own dom system allowed me to create a new selector api (#77). The plan is to extract the dom stuff into a new `zenstruck/dom` package so it can be utilized by this package + others that need to make dom assertions.

/cc @weaverryan, @jwage, @nikophil, @OskarStark, when/if time permits, I'd appreciate if you could test this in any html-heavy apps you have that use browser. I found and fixed (and added tests) for several edge cases when testing this in my app.

TODO:
- [x] extract `zenstruck/dom`

### CODE REVIEW

File: src/Browser/Assertion/SameUrlAssertion.php [hunk 1]
```diff
@@ -18,18 +18,25 @@
  * @author Kevin Bond <kevinbond@gmail.com>
  *
  * @internal
+ *
+ * @phpstan-type PartsToMatch = array{
```
Comment (OskarStark) [inline src/Browser/Assertion/SameUrlAssertion.php]:
```suggestion
 * @phpstan-type PartsToMatch array{
```

shouldn't be needed

**Comment by nikophil** [comment]
I just saw this issue... sorry for the late reply 😅

It's been a while I have not worked on an "html" app (only json apis, or full messenger apps). Sorry I can't help

**Comment by OskarStark** [comment]
@kbond I currently don't have time, but you have access to one or two of my projects, so feel free to run them against this branch

**Comment by kbond** [comment]
Thanks Oskar, I'll try it out on your projects when I pick this PR up again.
